lightning/src/ln/outbound_payment.rs

```diff
@@ -1167,11 +1167,7 @@ impl OutboundPayments {
 						},
 						PendingOutboundPayment::InvoiceReceived { payment_hash, retry_strategy, .. } => {
 							let total_amount = route_params.final_value_msat;
-							let recipient_onion = RecipientOnionFields {
-								payment_secret: None,
-								payment_metadata: None,
-								custom_tlvs: vec![],
-							};
+							let recipient_onion = RecipientOnionFields::spontaneous_empty();
```

Comment:
Not sure I understand why we want to use `RecipientOnionFields::spontaneous_empty()`. And the previous commit message doesn't give any insight.

Reply:
I don't mind skipping this if you prefer, but I thought it would be nice to be more concise here since we're ~duplicating this variant's handling code as part of `StaticInvoiceReceived` handling. 

> And the previous commit message doesn't give any insight.

Oops, will fix.